test/integration/instance.test.js

```diff
@@ -1634,6 +1635,39 @@ describe(Support.getTestDialectTeaser('Instance'), function() {
         expect(user.get({ plain: true })).to.deep.equal({ username: 'foo', id: null });
       });
     });
+
+    it('gets plain object with only requested attributes', function() {
+      var attrs = ['id', 'title'];
+      var ConnectionPlan = current.define('ConnectionPlan', {
+          title : DataTypes.TEXT,
+          price : DataTypes.INTEGER,
+          formattedPrice : {
+            type : DataTypes.VIRTUAL,
+             get : function () {
+               return '$' + this.getDataValue('price');
+             }
+          }
+      });
+
+      return ConnectionPlan.sync({force : true})
+        .then(function () {
+          return ConnectionPlan.create({
+            title : 'Lite Plan',
+            price : 100
+          });
+        })
+        .then(function () {
+          return ConnectionPlan.find({
+            attributes : attrs
+          });
+        })
+        .then(function (plan) {
+          var plain = plan.get({plain: true}),
+            keys = Object.keys(plain);
+
+          expect(!!_.difference(keys, attrs).length).to.equal(false);
```

Comment:
Not sure I agreed with this test - it would seem to suggest you have to add `formattedPrice` to `attributes` in order for it to be called. But if you do so, sequelize actually tries to select a field called `formattedPrice`.

I think we should only check if the key is in attributes if it is a normal attribute